compiler/ast/linter.nim

```diff
@@ -96,41 +178,44 @@ proc differ*(line: string, a, b: int, x: string): string =
     if cmpIgnoreStyle(y, x) == 0:
       result = y
 
-proc nep1CheckDefImpl(conf: ConfigRef; info: TLineInfo; s: PSym; k: TSymKind) =
+proc differs(conf: ConfigRef; info: TLineInfo; newName: string): string =
+  let line = sourceLine(conf, info)
+  var first = min(info.col.int, line.len)
+  if first < 0: return
+  #inc first, skipIgnoreCase(line, "proc ", first)
+  while first > 0 and line[first-1] in Letters: dec first
+  if first < 0: return
+  if first+1 < line.len and line[first] == '`': inc first
+
+  let last = first+identLen(line, first)-1
+  result = differ(line, first, last, newName)
+
+proc checkDefImpl(conf: ConfigRef; info: TLineInfo; s: PSym; k: TSymKind) =
   # operators stay as they are:
   if k in {skResult, skTemp} or s.name.s[0] notin Letters: return
   if k in {skType, skGenericParam} and sfAnon in s.flags: return
   if s.typ != nil and s.typ.kind == tyTypeDesc: return
   if {sfImportc, sfExportc} * s.flags != {}: return
-  if optStyleCheck notin s.options: return
+  if k == skParam and {sfImportc, sfExportc} * s.owner.flags != {}: return
+  if optStyleCheck notin conf.options: return
```

Comment:
I haven't tested it, but wouldn't the change here cause the following to fail (assuming that style checks are enabled globally):
```nim
{.push styleChecks: off.}

proc p[T]() =
  var Illegal = 1 # non-conforming name according to the style rules

{.pop.}

# instantiate:
p[int]()
```

Reply:
True, I found lots of symbols seem to lose style checks and didn't take the time to debug those. I'll revert this and might have to disable the test until it's fixed for good.